src/utils/helpers.ts

```diff
@@ -2,6 +2,19 @@ import { utils } from 'ethers';
 import { BigNumber } from './bignumber';
 import { MAX_UINT } from './constants';
 
+type EnumType = { [s: number]: string };
+
+export function mapEnum(enumerable: EnumType, fn: Function): any[] {
+  // get all the members of the enum
+  let enumMembers: any[] = Object.keys(enumerable).map(key => enumerable[key]);
+
+  // we are only interested in the numeric identifiers as these represent the values
+  let enumValues: number[] = enumMembers.filter(v => typeof v === 'number');

```

Comment:
Bit confused why we need to filter numbers out, is there anything else in there?

Reply:
updated the functions comments to clear this up.